### ISSUE TITLE
How to show orders and trades with a custom contract/socket?

### ISSUE THREAD

**Issue by bitcoin4cashqc** [etherdelta/etherdelta.github.io#284]
I installed the source on my server and deployed all the contracts needed on ropsten. Only issue I have is I can't see order between my test token and eth, as well most of the panel data. I see both balances tho!

Any hints to make the socket part custom and locally? :)

![capture du 2017-11-26 07-43-16](https://user-images.githubusercontent.com/17071641/33240063-94490ce6-d27d-11e7-9454-1dadf35a9db6.png)

**Comment by CineTek** [comment]
The socket part is just there for users for not having to mess around with web3 and the blockchain itself. If you have already established some knowledge with web3 it makes sense to track your orders and transactions using the blockchain instead of having to rely on a slow and not necessarily up to date socket backend.
This backend is not open source as far as I know so I highly recommend to write your own using transaction trackers like etherscan or ethplorer

**Comment by bitcoin4cashqc** [comment]
yeah sure, that's what I thougt. I have no issue doing on my own, I was just expecting that I could make it work on ropsten straigh forward. Trying to understand the contract process is a mess in remix/mist, since I assume posting order rely on the socket as well (Right)?

________________________________
De : CineTek <notifications@github.com>
Envoyé : 26 novembre 2017 13:44:01
À : etherdelta/etherdelta.github.io
Cc : bitcoin4cashqc; Author
Objet : Re: [etherdelta/etherdelta.github.io] How to show orders and trades with a custom contract/socket? (#284)


The socket part is just there for users for not having to mess around with web3 and the blockchain itself. If you have already established some knowledge with web3 it makes sense to track your orders and transactions using the blockchain instead of having to rely on a slow and not necessarily up to date socket backend.
This backend is not open source as far as I know so I highly recommend to write your own using transaction trackers like etherscan or ethplorer

**Comment by bitcoin4cashqc** [comment]
issue is not missing knowlegde with web3, it's with socket.io :P

________________________________
De : CineTek <notifications@github.com>
Envoyé : 26 novembre 2017 13:44:01
À : etherdelta/etherdelta.github.io
Cc : bitcoin4cashqc; Author
Objet : Re: [etherdelta/etherdelta.github.io] How to show orders and trades with a custom contract/socket? (#284)


The socket part is just there for users for not having to mess around with web3 and the blockchain itself. If you have already established some knowledge with web3 it makes sense to track your orders and transactions using the blockchain instead of having to rely on a slow and not necessarily up to date socket backend.
This backend is not open source as far as I know so I highly recommend to write your own using transaction trackers like etherscan or ethplorer

**Comment by bitcoin4cashqc** [comment]
I guess this may be intersting for anyone looking to make etherdelta work locally with custom contract/tokens : 

https://github.com/beaquant/EtherDelta.io

**Comment by 360disrupt** [comment]
I think without the sockets you can't the Order Events are not broadcasted because the orders are currently off-chain.
